File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.empty()) return 0;

        int k = 1; // Pointer for unique elements
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] != nums[i - 1]) { // Found a new unique element
                nums[k] = nums[i]; // Move it to the front
                k++;
            }
        }
        return k;
    }
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0

        k = 1 # Pointer for the unique elements index
        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]: # If a new unique element is found
                nums[k] = nums[i] # Move it to the front
                k += 1 # Increment unique elements count
        
        return k
        